File: app/src/main/java/info/nukoneko/cuc/android/kidspos/repository/StaffManager.java
package info.nukoneko.cuc.android.kidspos.repository;

import android.content.Context;
import android.support.annotation.NonNull;

import info.nukoneko.cuc.android.kidspos.api.APIAdapter;
import info.nukoneko.cuc.android.kidspos.entity.Staff;
import io.reactivex.Observable;

public final class StaffManager extends BaseManager {
    public StaffManager(@NonNull Context context, @NonNull APIAdapter apiAdapter) {
        super(context, apiAdapter);
    }

    public Observable<Staff> getStaff(String barcode) {
        return getAPIService().getStaff(barcode);
    }
}

File: app/src/main/kotlin/info/nukoneko/cuc/android/kidspos/manager/StaffManager.kt
package info.nukoneko.cuc.android.kidspos.manager

import info.nukoneko.cuc.android.kidspos.api.APIAdapter
import info.nukoneko.cuc.android.kidspos.entity.Staff
import io.reactivex.Single

class StaffManager(apiAdapter: APIAdapter) : BaseManager(apiAdapter) {
    fun getStaff(barcode: String): Single<Staff> {
        return apiService.getStaff(barcode)
    }
}
